### ISSUE TITLE
Tree: Separate button icon to expand/collapse node + node children count awareness & display

### ISSUE THREAD

**Issue by hemantg05** [infor-design/enterprise#3847]
As shown in the screenshot below, we would like to have a separate icon which can be clicked to collapse/expand the node and the support for children count feature. In addition we would like to support an additional icon.

<img width="270" alt="tree_node_separate_expand_collapse_button_icon" src="https://user-images.githubusercontent.com/1594815/80581921-31543980-8a0e-11ea-85cc-5b726122bd62.png">

**Describe the solution you'd like**

**Expand/Collapse (+/-) icons:**
- A configuration to show/hide separate button icons for expand/collapse.
- The expand/collapse icon should not be shown for a leaf node
- Click on expand/collapse button icon should only expand/collapse the tree node.
- Click on expand/collapse button should NOT select the tree node.
- Good to have: The possibility to use different expand/collapse icons

**Count:**
- The children are fetched when user clicks on the node. We will know the children count but won't have the exact children available until user clicks on the node and the server returns the children. So, tree should have some mechanism to show expand/collapse icon on the basis of count. 
- Each tree node should have some property like count of children (auto calculated on the basis of actual children) or directly set via a tree node property. The count label is shown beside the name. This is similar to "N results" feature on datagrid.
- The label automatically changes if the number of children changes (by addition or removal) if the count is auto-calculated instead of manually being set via tree node property.

**Angular support:**
- The feature should also be available in enterprise-ng

**Additional context**
Infor PIM
Product Owner: @karinthulin

**Comment by tmcconechy** [comment]
Part of this request is on https://github.com/infor-design/enterprise/issues/3730

**Comment by CindyMercadoReyes** [comment]
Per my conversation with Tim, this ticket is still not ready for QA. The change is not being reflexed  yet.

**Comment by jbrcna** [comment]
**QA Passed**
v4.30.0-rc0
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-ajax.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-sortable.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-markup.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-badges.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-select-multiple.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-edit-auto.html
https://4300-rc0-enterprise.demo.design.infor.com/components/tree/test-toggle-icon-and-count-edit-manually.html
